### DISCUSSION TITLE
Tracking in a 3D space

### DISCUSSION THREAD

**Discussion by reneemapa** [talmolab/sleap#1155] [General]
Hello!

I am looking into sleap as a potential machine learning tracking algorithm to label and track my data. I work with fish and we take videos of the tanks from the side angle, so in a 3D space. In these videos, there are two fish present and they often swim past each other and may overlap. In the past we have had difficulties finding an algorithm that successfully tracks the fish without continuously swapping their identities. I was wondering if anyone is familiar with how sleap manages regarding tracking in 3D spaces.

Here is an example of the setup as well as an instance where the fish may overlap. 

![Aggression](https://user-images.githubusercontent.com/71779894/215127571-84d34443-221a-450e-9915-81b3f203595a.png)
![Aggression2](https://user-images.githubusercontent.com/71779894/215127574-5e47a564-3622-4258-881e-0467b6267e8f.png)

Thanks all!

**Comment by roomrys** [comment]
Hi @reneemapa,

We have recently gotten a lot of questions on how to improve tracking (#1126 and #1149). SLEAP also has trouble with mistaken identities when occlusion occurs - we are doing some case by case trials with supervised ID, but this is still very experimental. If you are interested in trying out our supervised ID tracking, get in touch with me at [lmaree@salk.edu](mailto:lmaree@salk.edu). Hopefully some other members in the community can give you their candid and constructive feedback as well.

Thanks,
Liezl
